reefd/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type machineConfigInput struct {
+	Name          string `json:"name"`
+	Configuration string `json:"configuration"`
+}
+
+type machineConfig struct {
+	id            int
+	name          string
+	configuration string
+}
+
+// handleGetRequest processes GET request to /machine/configuration that query the machine config based on name
+func handleGetRequest(db *sql.DB, w http.ResponseWriter, r *http.Request) {
```

Comment:
could you define a server struct/class, with db wrapped in it? and make this a method of that server?